app/test_main.py

```diff
@@ -1 +1,28 @@
-# write your code here
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+url = "https://www.notavalidurl.com"
+
+
+def test_can_access_google_page_with_valid_input() -> None:
+    with patch("app.main.has_internet_connection", return_value=True), (
+         patch("app.main.valid_google_url", return_value=True)):
```

Comment:
Imagine having a fixture and not writing those patches again and again..

Reply:
Sorry but I don`t understand where should I fix it